Marlin/src/pins/pins_PRINTRBOARD_REVF.h

```diff
@@ -169,23 +169,26 @@
 #endif
 
 #if ENABLED(MINIPANEL)
-  #if ENABLED(USE_INTERNAL_SD)
-    //      PIN       FASTIO PIN#  ATUSB90 PIN# Teensy2.0++ PIN#
-    #define SDSS               20  //        10               B0
-    #define SD_DETECT_PIN      -1  // no auto-detect SD insertion on built-in Printrboard SD reader
-  #else
+  #if DISABLED(USE_INTERNAL_SD)
     //      PIN       FASTIO PIN#  ATUSB90 PIN# Teensy2.0++ PIN#  Printrboard RevF Conn.   MKSLCD12864 PIN#
+    #undef SDSS
```

Comment:
This pins file is standalone, so `#undef` should not be needed. Instead of using `#undef` here and allowing the values to be set above, apply orderly conditionals to set the pins to the desired values. Note that several pins can be left out if they're only going to be set to `-1`. A quick scan shows that you can get rid of the `SD_DETECT_PIN` and `BEEPER_PIN` set to -1 above, for example, removing the need to undefine them below. Others, like `BTN_ENC`, should simply not be set above. Remember that `MINIPANEL` sets `ULTIPANEL` sets `ULTRA_LCD`.

The main issue with this pins file is just not understanding the above points.

Reply:
The purpose of the PR is the Teensyduino, so I'll remove the pins changes (I was tagging them on) and revisit how they are supposed to work).

Reply:
> Remember that MINIPANEL sets ULTIPANEL sets ULTRA_LCD

The relationship between ULTIPANEL and ULTRA_LCD wasn't clear.  And when I originally attempted to implement MINIPANEL with Bob, I thought MINIPANEL was setting the pins upstream but that was when I first started using Marlin so wasn't as familiar.  This makes more sense.

My confusion around the pins treatment [reminds me of this discussion with Roxy about NEWPANEL, ULTIPANEL, ULTRA_LCD, etc](https://github.com/MarlinFirmware/Marlin/issues/6982)